### ISSUE TITLE
Two-Line preview for compact room list mode

### ISSUE THREAD

**Issue by BurnyBoi** [SchildiChat/schildichat-desktop#140]
I really like the compact mode for the room list, and is one of the main things that makes me use SchildiChat over Element. However, I don't like that the only option for compact mode is to have the room preview in one-line mode. I have a vertical monitor and need to keep the room list narrow, so I can't read the room previews in compact mode at all.

While having two-line mode would make compact mode slightly-less compact, I think it would be beneficial to have it as an option. My recommendation would be to break out the preview mode into it's own setting, and let the user select one-line or two-line mode separate from the room list style.

**Comment by su-ex** [comment]
Adding such an option would double the amount of variants effectively available. This is nothing I can test and maintain in the long run. Also, it's complicated enough how I weaved in the existing options in what Element provided as base.
It's probably easier to add yet another variant additionally to those three available right now. But I can't quite grasp how what you want it's different compared to the intermediate option, can you maybe provide a sketch?

**Comment by SpiritCroc** [comment]
I'd also say it makes sense to not have separate settings, since I assume for most users it should make sense to change both line count and avatar size at the same time, and having to touch two settings to achieve a sensible layout is more complicated.

**Comment by BurnyBoi** [comment]
>But I can't quite grasp how what you want is different compared to the intermediate option, can you maybe provide a sketch?

Intermediate forces every room on the list to take up what looks like twice as much room as Compact because it makes the avatars bigger. If Compact had the preview on a separate line, I would still get the benefit of having more rooms on my screen compared to Intermediate, especially if I only enable previews on my favorites (which is what I usually do)

**Comment by su-ex** [comment]
I still don't understand what you want.
Do you want room list tile items with different height depending on if the preview is shown? But what's the benefit in not making the avatar bigger as well in that case to match both lines in height?

**Comment by BurnyBoi** [comment]
>Do you want room list tile items with different height depending on if the preview is shown?
Yes. This would be necessary to fit the preview underneath the name.

>But what's the benefit in not making the avatar bigger as well in that case to match both lines in height?
The avatar isn't relevant. I could take or leave making it bigger in this scenario, whatever's easier. The point is that the rooms with no preview stay compact. Basically, I set Favorites/People to have previews, my main room list stays compact, I have more rooms on screen overall vs Intermediate mode.

**Comment by su-ex** [comment]
Ouch, that is something which can't be implemented easily, because Element has hardcoded the tile heights for some reason. Thus the whole room list would be need re-implemented in a way this can be achieved. This is nothing I will do right now, but if Element ever restructures the room list so that I have to redo all options anyway, I might remember this.
